src/components/textbox/textbox.test.js

```diff
@@ -0,0 +1,712 @@
+          .parent()
+          .should("have.css", "width", inputRatio);
+      }
+    );
+
+    it("should render Textbox with required prop", () => {
+      CypressMountWithProviders(<TextboxComponent required />);
+
+      verifyRequiredAsteriskForLabel();
+    });
+
+    it.each([
+      ["11", "11", "rgba(0, 0, 0, 0.55)"],
+      ["11", "10", "rgb(203, 55, 74)"],
+    ])(
+      "should input %s characters and warn if over character limit of %s in Textbox",
+      (charactersUsed, limit, color) => {
+        const inputValue = "12345678901";
+
+        CypressMountWithProviders(
+          <TextboxComponent
+            enforceCharacterLimit={false}
+            warnOverLimit
+            characterLimit={limit}
+          />
+        );
+
+        textboxInput()
+          .type(inputValue)
+          .then(() => {
+            commonInputCharacterLimit()
+              .should("have.text", `${charactersUsed}/${limit}`)
+              .and("have.css", "color", color);
+          });
+      }
+    );
+
+    it.each([
+      ["11", "11"],
+      ["10", "10"],
+    ])(
+      "should input %s characters and enforce character limit of %s in Textbox",
+      (charactersUsed, limit) => {
+        CypressMountWithProviders(
+          <TextboxComponent enforceCharacterLimit characterLimit={limit} />
+        );
+
+        const inputValue = "12345678901";
+
+        textboxInput()
+          .type(inputValue)
+          .then(() => {
+            commonInputCharacterLimit().should(
+              "have.text",
+              `${charactersUsed}/${limit}`
+            );
+          });
+      }
+    );
+
+    it("should render Textbox with name prop", () => {
+      CypressMountWithProviders(
+        <TextboxComponent name={CHARACTERS.STANDARD} />
+      );
+
+      textboxInput().should("have.attr", "name", CHARACTERS.STANDARD);
+    });
+
+    it("should render Textbox with disabled prop", () => {
+      CypressMountWithProviders(<TextboxComponent disabled />);
+
+      textboxInput().should("be.disabled").and("have.attr", "disabled");
+    });
+
+    it.each([CHARACTERS.SPECIALCHARACTERS, CHARACTERS.DIACRITICS])(
+      "should render Textbox with placeholder prop set to %s",
+      (placeholder) => {
+        CypressMountWithProviders(
+          <TextboxComponent placeholder={placeholder} />
+        );
+
+        textboxInput().should("have.attr", "placeholder", placeholder);
+      }
+    );
+
+    it("should render Textbox with autoFocus prop", () => {
+      CypressMountWithProviders(<TextboxComponent autoFocus />);
+
+      body().tab();
+      textboxInput()
+        .should("be.focused")
+        .then(($el) => {
+          checkGoldenOutline($el.parent());
+        });
+    });
+
+    it("should render Textbox with readOnly prop", () => {
+      CypressMountWithProviders(<TextboxComponent readOnly />);
+
+      textboxInput().and("have.attr", "readOnly");
+      textbox().should("have.attr", "readOnly");
```

Comment:
I would remove this line. The important thing is the `readonly` attribute on the input - the fact that this also appears on the containing `div` is just an accident and doesn't actually do anything.